### PR TITLE
compute: remove mz_cluster_replica_heartbeats table

### PR DESCRIPTION
**Pull request by benesch**
This table was added back in 2022 to provide visibility into replica health.  It hasn't proven to be useful in practice: nothing in our console nor our cloud monitoring depends on the `last_heartbeat` value. In particular, the visibility into a replica's dataflow frontiers, provided by `mz_cluster_replica_frontiers`, has proven to be much more useful than `mz_cluster_replica_heartbeats`.

So, this commit removes the `mz_cluster_replica_heartbeats` table. This is one less collection to worry about during 0dt upgrades (#22567).

### Motivation

   * This PR removes existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/compute-client/src/controller/instance.rs [hunk 1]
```diff
@@ -341,22 +340,7 @@ impl<T: ComputeControllerTimestamp> Instance<T> {
...
-            ]);
-            self.deliver_introspection_updates(
-                IntrospectionType::ComputeReplicaHeartbeats,
-                vec![(row, -1)],
-            );
-        }
-
-        Some(replica)
+        self.replicas.remove(&id)
     }
```
Comment (teskje) [inline src/compute-client/src/controller/instance.rs:344]: Nit: Now that this method only consists of a single line, how about inlining it in the only place where it is used?

**Review by teskje — APPROVED** [review]
Thanks! Compute changes lgtm.

File: src/compute-client/src/controller/instance.rs (hunk 1 above)
Comment (benesch) [reply src/compute-client/src/controller/instance.rs]: Works for me!

**Comment by benesch** [comment]
Thanks for the review, @teskje! Marking for automerge.

**Pull request merged**